Source/WebGPU/WebGPU/RenderPassEncoder.h

```diff
@@ -98,6 +98,7 @@ class RenderPassEncoder : public WGPURenderPassEncoderImpl, public RefCounted<Re
     uint64_t m_debugGroupStackSize { 0 };
 
     const Ref<Device> m_device;
+    RefPtr<const RenderPipeline> m_renderPipeline;
```

Comment:
In order to better be able to reason about object lifetime, I think we should pull out the data we need from the render pipeline and retain that, instead of retaining the render pipeline itself.

Reply:
Completely agree. I'll change that